packages/paste-core/components/editable-code-block/src/EditableCodeBlock.tsx

```diff
@@ -6,34 +6,149 @@ import {
-    // Sets the Paste theme for the editor
-    monaco.editor.defineTheme('paste', CodeEditorPasteTheme);
-    monaco.editor.setTheme('paste');
-
-    /*
-     * Call provided onMount function.
-     * This can be used to add custom language support, or to grab a
-     * reference to the ref (monaco) object.
-     */
-    onMount?.(editor, monaco);
-  };
+  const controlledOptions = React.useMemo(() => {
+    return {
+      lineNumbers,
+      folding,
+      readOnly,
+      scrollBeyondLastLine,
+      glyphMargin: inlineErrorRange != null,
+      // ...monacoOptions,
+      minimap: {
+        enabled: showMinimap,
+        // ...monacoOptions.minimap,
+      },
+      guides: {
+        indentation: indentationGuide,
+        // ...monacoOptions.guides,
+      },
+    } as CodeEditorProps['options'];
+  }, [
+    // monacoOptions,
+    readOnly,
+    showMinimap,
+    folding,
+    inlineErrorRange,
+    scrollBeyondLastLine,
+    indentationGuide,
+    lineNumbers,
+  ]);
+
+  const handleEditorDidMount = React.useCallback(
+    (editor: Editor.IStandaloneCodeEditor, monaco: Monaco): void => {
+      // Sets the Paste theme for the editor
+      monaco.editor.defineTheme('paste', CodeEditorPasteTheme);
+      monaco.editor.setTheme('paste');
+
+      if (inlineErrorRange) {
+        editor.createDecorationsCollection([
+          {
+            range: new monaco.Range(
+              inlineErrorRange.startLineNumber,
+              inlineErrorRange.startColumn,
+              inlineErrorRange.endLineNumber,
+              inlineErrorRange.endColumn
+            ),
+            options: {
+              isWholeLine: inlineErrorIsWholeLine,
+              className: '.paste-editable-code-editor-row-error',
+              hoverMessage: inlineErrorHoverMessage,
+              glyphMarginClassName: '.paste-editable-code-editor-margin-error',
+              glyphMarginHoverMessage: inlineErrorHoverMessage,
+            },
+          },
+        ]);
+      }
+
+      /*
+       * Call provided onMount function.
+       * This can be used to add custom language support, or to grab a
+       * reference to the ref (monaco) object.
+       */
+      onMount?.(editor, monaco);
+    },
+    [onMount, inlineErrorHoverMessage, inlineErrorRange, inlineErrorIsWholeLine]
+  );
 
   return (
     <Box element={element} borderRadius="borderRadius10" overflow="hidden">
-      <CodeEditor {...props} onMount={handleEditorDidMount} />
+      {inlineErrorRange ? (
+        <StylingGlobals
+          styles={{
+            '.paste-editable-code-editor-margin-error': {
+              ':before': {
+                content: '""',
+                display: 'inline-block',
+                width: '16px',
+                height: '16px',
+                background: `url(https://assets.twilio.com/public_assets/paste-assets/1.0.0/icons/editable-codeblock-error.svg) no-repeat center center`,
```

Comment:
suggestion: Any chance the icon could be moved 4px to the right so it's not right up against the edge of the code block?

Reply:
mmmmk 2px will do 👍👍